libs/prov/prov_k8s_cortx_deploy.py

```diff
@@ -1728,3 +1723,44 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
         else:
             LOGGER.info("Installing version is not higher than installed version.")
         return installing_version
+
+    def update_sol_for_granular_deploy(self, file_path: str, host_list: list,
+                                       image: str, deployment_type: str) -> tuple:
+        """
+        Helper function to update image in solution.yaml.
+        :param: file_path: Filename with complete path
+        :param: host_list: List of setup hosts
+        :param: image: Image to be used for deployment
+        :param: deployment_type: Type of deployment(Standard/Data-Only)
+        :return: True/False and local file
+        """
+        LOGGER.debug("Update nodes section with setup details.")
+        prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        resp = self.update_nodes_sol_file(file_path, host_list)
+        if not resp[0]:
+            return False, "solution.yaml is not updated properly."
+        LOGGER.debug("Update storage section and deployment type.")
+        with open(file_path) as soln:
+            conf = yaml.safe_load(soln)
+            parent_key = conf['solution']
+            storage_key = parent_key["storage"]
+            soln.close()
+        parent_key["deployment_type"] = deployment_type
+        if "data" in deployment_type:
+            parent_key["images"]["cortxdata"] = image
+        else:
+            # Extend else condition when a different granular deployment type is available.
+            pass
+        for cvg in prov_deploy_cfg["cvg_config"]:
+            if cvg == "cvg1":
+                cvg_key = storage_key[cvg]["devices"]["data"]
+                cvg_key.pop("d7")
```

Comment:
what if i need to have 10 disk per cvg in that case will it work?

Reply:
Please use generic forms so that we can avoid hardcoded values

Reply:
Done.